src/main/kotlin/com/terraformation/backend/file/GoogleDriveWriter.kt

```diff
@@ -151,16 +156,72 @@ class GoogleDriveWriter(
   ): String {
     return fileStore.read(metadata.storageUrl).use { inputStream ->
       uploadFile(
-          driveId = driveId,
           parentFolderId = parentFolderId,
           filename = name,
           contentType = metadata.contentType,
           inputStream = inputStream,
+          driveId = driveId,
           description = description,
           fileId = metadata.id)
     }
   }
 
+  /**
+   * Looks up the drive ID for a given file ID. The file may be a folder or a regular file.
+   *
+   * @throws NoSuchFileException The requested file wasn't found. This may mean the file truly
+   *   doesn't exist, or it may mean that the user doesn't have permission to see the file.
+   */
+  fun getDriveIdForFile(googleFileId: String): String {
+    return getFileMetadata(googleFileId).driveId
+  }
+
+  /**
+   * Extracts the file ID from a Google Drive folder URL. The file ID is assumed to be the last path
+   * element in the URL.
+   */
+  fun getFileIdForFolderUrl(folderUrl: URI): String {
+    if (folderUrl.host == "drive.google.com" && folderUrl.path.startsWith("/drive/folders/")) {
+      return folderUrl.path.substringAfterLast('/')
+    } else {
+      throw IllegalArgumentException("$folderUrl does not appear to be a Google Drive folder URL")
+    }
+  }
+
+  fun renameFile(googleFileId: String, newName: String) {
+    val newMetadata = File()
+    newMetadata.name = newName
+    val metadata = getFileMetadata(googleFileId)
+    metadata.id = null
+    metadata.name = newName
```

Comment:
Are we callin `getFileMetadata` as a check for file existence? Curious if we need lines 195 and 196.

Reply:
Ah, good catch. This was left over from an earlier version of the function. Will remove it.